### PR TITLE
PS-4839 : Adjust MyRocks and TokuDB defaults for 8.0

### PR DESCRIPTION
**Pull request by george-lorch**
- Changed default value of the rocksdb_default_cf_options to be :
        block_based_table_factory={
            cache_index_and_filter_blocks=1;
            filter_policy=bloomfilter:10:false;
            whole_key_filtering=1};
        level_compaction_dynamic_level_bytes=true;
        optimize_filters_for_hits=true;
        compaction_pri=kMinOverlappingRatio;
        compression=kLZ4Compression;
        bottommost_compression=kLZ4Compression;
- Changed default rocksdb_max_total_wal_size from rocksdb default(0) to be 2G
- Changed default rocksdb_block_size from rocksdb default(4K) to 16K
- Changed/enforced existing default of rocksdb_table_cache_numshardbits to
  current rocksdb default of 6
- Changed/enforced existing default of rocksdb_max_background_jobs to
  current rocksdb default of 2
- Changed default tokudb_enable_partial_eviction to be false.
- Changed default tokudb_block_size to be 512M.
- Changed default tokudb_read_block_size to be 16K.
- Changed default tokudb_row_format to be quicklz.
- Fixed and re-recorded tests impacted by above changes:
  - rocksdb.mysqldump2 - new block_size (4x bigger) changes value range of
    rocksdb status variable rocksdb_block_cache_add
  - rocksdb.index_merge_rocksdb - new block size changes optimizer conditions
    and causes different query plan within test for a specifically targeted
    EXPLAIN UPDATE.  Revert to old block size for this test.
  - rocksdb.compact_deletes - new block size changes WAL and SST metrics and
    causes different compaction behavior from what the test expects.  Revert
    to old block size for this test.
  - rocksdb.rocksdb - re-recorded new defaults.
  - rocksdb.block_size - re-recorded new defaults.
  - rocksdb.sys_vars.rocksdb_block_size_basic - re-recorded new defaults.
  - rocksdb.sys_vars.rocksdb_default_cf_options_basic - re-recorded new
    defaults.
  - rocksdb.sys_vars.rocksdb_max_total_wal_size_basic - re-recorded new
    defaults.
  - tokudb.bugs.simple_icp - new tree shapes produce slightly different metrics
    to optimizer which change some EXPLAIN results, re-recorded.

### CODE REVIEW

**Comment by george-lorch** [comment]
https://ps.cd.percona.com/view/8.0/job/percona-server-8.0-param/80/

File: storage/rocksdb/ha_rocksdb.cc [hunk 1]
```diff
@@ -412,6 +412,10 @@ static constexpr ulong RDB_MAX_LOCK_WAIT_SECONDS = 1024 * 1024 * 1024;
 static constexpr ulong RDB_MAX_ROW_LOCKS = 1024 * 1024 * 1024;
 static constexpr ulong RDB_DEFAULT_ROW_LOCKS = 1024 * 1024;
 static constexpr ulong RDB_DEFAULT_BULK_LOAD_SIZE = 1000;
+static constexpr int RDB_DEFAULT_MAX_BACKGROUND_JOBS = 2;
```
Comment (laurynas-biveinis) [inline storage/rocksdb/ha_rocksdb.cc:415]: Strictly speaking each "constexpr" has to be "const constexpr" here but meh

**Review by laurynas-biveinis — APPROVED** [review]
It would be logical to split Rocks and Toku default changes to separate commits, but meh

File: storage/rocksdb/ha_rocksdb.cc (hunk 1 above)
Comment (george-lorch) [reply storage/rocksdb/ha_rocksdb.cc:415]: true, I'll JIRA this to review these as well as TokuDB later and fix them up.
https://jira.percona.com/browse/PS-4964

**Pull request merged**